torchgeo/datasets/usavars.py

```diff
@@ -0,0 +1,227 @@
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            data and label at that index
+        """
+        sample = self.files[index]
+        tensor_sample = {}
+        tensor_sample["image"] = self._load_image(sample["image"])
+
+        keys = [key for key in sample.keys() if key != "image"]
+        for key in keys:
+            tensor_sample[key] = Tensor([sample[key]])
+
+        if self.transforms is not None:
+            tensor_sample = self.transforms(tensor_sample)
+
+        return tensor_sample
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.files)
+
+    def _load_files(self) -> List[Dict[str, Any]]:
+        import pandas as pd
+
+        file_path = os.path.join(self.root, self.dirname, "uar")
+        files = os.listdir(file_path)
+
+        files = files[
+            :10
+        ]  # TODO: remove this, keeping temporarily because this func is very slow
+
+        # csvs = self.label_urls.keys() # only uar for now
+        csvs = ["treecover", "elevation", "population"]
+        labels_ds = [
+            (lab, pd.read_csv(os.path.join(self.root, lab + ".csv"))) for lab in csvs
+        ]
+        samples = []
+        for f in files:
+            img_path = os.path.join(file_path, f)
+            samp = {"image": img_path}
+
+            id_ = f[5:-4]
+
+            for lab, ds in labels_ds:
+                samp[lab] = ds[ds["ID"] == id_][lab].values[0]
+
+            samples.append(samp)
+        return samples
+
+    def _load_image(self, path: str) -> Tensor:
+        with rasterio.open(path) as f:
+            array: "np.typing.NDArray[np.int_]" = f.read()
+            tensor: Tensor = torch.from_numpy(array)  # type: ignore[attr-defined]
+            return tensor
+
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset.
+
+        Raises:
+            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
+        """
+        # Check if the extracted files already exist
+        pathname = os.path.join(self.root, self.dirname)
+        if glob.glob(pathname):
+            return
+
+        # Check if the zip files have already been downloaded
+        pathname = os.path.join(self.root, self.zipfile)
+        if glob.glob(pathname):
+            self._extract()
+            return
+
+        # Check if the user requested to download the dataset
+        if not self.download:
+            raise RuntimeError(
+                f"Dataset not found in `root={self.root}` and `download=False`, "
+                "either specify a different `root` directory or use `download=True` "
+                "to automaticaly download the dataset."
+            )
+
+        self._download()
+        self._extract()
+
+    def _download(self) -> None:
```

Comment:
When I downloaded this with `USAVars(root="data/", download=True)`, then the directory structure looks like:
`data/housing.csv`
...
`data/uar.zip`
`data/uar/*.tif`

However, `_load_files()` expects the files to be in a different place

Reply:
This happens because I decided to name the zipfile usavars. So you download usavars.zip which has uar inside of it. The reasoning behind this is that we're planning to add POP so then you'd have usavars/{uar, pop} or something. If we just want everything in one directory it's an easy fix

Reply:
this line `file_path = os.path.join(self.root, self.dirname, "uar")` is a bug regardless

Reply:
I.e. `_load_files()` is only working for you because of the way you have manually setup your dataset I think